### PR TITLE
NEXT-26196 - Add support for active flag on user entity

### PR DESCRIPTION
**Pull request by t2oh4e**
### 1. Why is this change necessary?
Active flag on user entity is currently not considered, but can be set via API. This PR adds the field to the admin ui and uses it to handle user authentication.

### 2. What does this change do, exactly?
Add user active field to admin ui. Consider the field during authentication.

### 3. Describe each step to reproduce the issue or behaviour.
Create admin user.
Set active to 0 via api or database.
Log in to admin with said user.

### 4. Please link to the relevant issues (if any).
https://issues.shopware.com/issues/NEXT-26196

### 5. Checklist

- [x] I have rebased my changes to remove merge conflicts
- [ ] I have written tests and verified that they fail without my change
- [x] I have created a [changelog file](https://github.com/shopware/platform/blob/trunk/adr/2020-08-03-implement-new-changelog.md) with all necessary information about my changes
- [ ] I have written or adjusted the documentation according to my changes
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfil them.

### <samp>🤖 Generated by Copilot at 7a55059</samp>

This pull request adds support for the active flag of the user entity, which allows administrators to enable or disable users in the system. It adds a new switch field in the user detail page of the administration module, and modifies the API and OAuth logic to check the user's active status. It also updates the changelog and the translations for the new feature.

### <samp>🤖 Generated by Copilot at 7a55059</samp>

*  Add support for the active flag of the user entity ([link](https://github.com/shopware/platform/pull/3048/files?diff=unified&w=0#diff-d5be76f7ec7c85b0dd808eeaa902226dd73ee41ef28014e239a8d3fdb29922b8L1-R36))
*  Add a switch field for the active flag in the user detail page of the administration module ([link](https://github.com/shopware/platform/pull/3048/files?diff=unified&w=0#diff-6d7ee0888379e78a5fb1ef243a56d22bafad6023e5c2de269fdc4d94a89c81e8L191-R213), [link](https://github.com/shopware/platform/pull/3048/files?diff=unified&w=0#diff-e2f509af8d5821c213f1aa6d5e5bc77c3af52e397a54d30d079c56e330c977eeL73-L76), [link](https://github.com/shopware/platform/pull/3048/files?diff=unified&w=0#diff-f75976b68ea055916512b199e30923dfc0346815363faa2ada4c6b4de5fbda4aR107), [link](https://github.com/shopware/platform/pull/3048/files?diff=unified&w=0#diff-0c13d8b292503c8c8de78a08b3dcb279b685f693602390d3d4d201163904f5c3R107))
*  Prevent inactive users from accessing the API with a valid token by checking the active flag in the `BearerTokenValidator` ([link](https://github.com/shopware/platform/pull/3048/files?diff=unified&w=0#diff-cbeb103eeed9cdf8e13969efbc6d032e78338b963c5459cf17a32338d9454e81R58))
*  Prevent inactive users from logging in with a valid password by checking the active flag in the `UserRepository` ([link](https://github.com/shopware/platform/pull/3048/files?diff=unified&w=0#diff-bda54bebf15b763d559cd5b9704cf0b66b0014b50b92fe9c98d5a9a58fbb5160R40))

### CODE REVIEW

File: changelog/_unreleased/2023-04-19-add-support-for-active-flag-of-user-entity.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+---
+title: Add support for active flag of user entity
+issue: NEXT-26196
+author: Timo Helmke
+author_email: t.helmke@kellerkinder.de
+author_github: t2oh4e
+---
+# Core
+*
```
Comment (mitelg) [inline changelog/_unreleased/2023-04-19-add-support-for-active-flag-of-user-entity.md]: would you mind changing this to something meaningful? 😁

Comment (t2oh4e) [reply changelog/_unreleased/2023-04-19-add-support-for-active-flag-of-user-entity.md]: Wooopsie 😅  Changed.

File: src/Administration/Resources/app/administration/src/module/sw-users-permissions/page/sw-users-permissions-user-detail/sw-users-permissions-user-detail.html.twig [hunk 2]
```diff
@@ -188,15 +188,29 @@
...
+                            {% endblock %}
+
+                            <!-- eslint-disable-next-line sw-deprecation-rules/no-twigjs-blocks -->
+                            {% block sw_settings_user_detail_grid_content_acl_is_active %}
+                            <sw-switch-field
+                                v-model="user.active"
+                                class="sw-settings-user-detail__grid-is-active"
+                                :label="$tc('sw-users-permissions.users.user-detail.labelActive')"
+                                :disabled="isCurrentUser || !acl.can('users_and_permissions.editor')"
+                            />
```
Comment (sydinh) [inline src/Administration/Resources/app/administration/src/module/sw-users-permissions/page/sw-users-permissions-user-detail/sw-users-permissions-user-detail.html.twig:211]: Would you mind adding a test for this added element 😉

**Comment by daothithientamm** [comment]
Hi @t2oh4e ,

It will be better if activate value are set true by default when create a new user. 
image

**Comment by shyim** [comment]
Hey,

I will close this PR as it's inactive for a while. Feel free to create a new one when you apply the requested changes

**Pull request closed without merging**